archethic/contracts/uco_pool.exs

```diff
@@ -266,6 +266,46 @@ actions triggered_by: transaction, on: reveal_secret(htlc_genesis_address, _evm_
   )
 end
 
+condition triggered_by: transaction, on: refund(htlc_genesis_address), as: [
+  content: (
+    # Ensure htlc_genesis_address exists in pool state
+    # and end_time has not been reached
+    valid? = false
+
+    contract_content = Json.parse(contract.content)
+
+    htlc_genesis_address = String.to_hex(htlc_genesis_address)
+    requested_secrets = Map.get(contract_content, "requested_secrets", Map.new())
+    htlc_map = Map.get(requested_secrets, htlc_genesis_address)
+
+    if htlc_map != nil do
+      valid? = htlc_map.end_time < Time.now()
```

Comment:
The comparison should be `<=` as the refund is triggered automatically at end_time